streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -1,1851 +1,1807 @@
-      cancelled <- substreamCancelled.get
-    } yield cancelled must_=== true
-  }
-
-  private def flatMapParSwitchInnerErrorsInterruptAllFibers = unsafeRun {
-    for {
-      substreamCancelled <- Ref.make[Boolean](false)
-      latch              <- Promise.make[Nothing, Unit]
-      result <- Stream(
-                 Stream.fromEffect((latch.succeed(()) *> UIO.never).onInterrupt(substreamCancelled.set(true))),
-                 Stream.fromEffect(latch.await *> IO.fail("Ouch"))
-               ).flatMapParSwitch(2)(identity).runDrain.either
-      cancelled <- substreamCancelled.get
-    } yield (cancelled must_=== true) and (result must beLeft("Ouch"))
-  }
-
-  private def flatMapParSwitchFinalizerOrdering = unsafeRun {
-    for {
-      execution <- Ref.make(List.empty[String])
-      inner     = Stream.bracket(execution.update("InnerAcquire" :: _))(_ => execution.update("InnerRelease" :: _))
-      _ <- Stream
-            .bracket(execution.update("OuterAcquire" :: _).as(inner))(_ => execution.update("OuterRelease" :: _))
-            .flatMapParSwitch(2)(identity)
-            .runDrain
-      results <- execution.get
-    } yield results must_=== List("OuterRelease", "InnerRelease", "InnerAcquire", "OuterAcquire")
-  }
-
-  private def flatMapParSwitchOuterErrorsInterruptAllFibers = unsafeRun {
-    for {
-      substreamCancelled <- Ref.make[Boolean](false)
-      latch              <- Promise.make[Nothing, Unit]
-      result <- (Stream(()) ++ Stream.fromEffect(latch.await *> IO.fail("Ouch")))
-                 .flatMapParSwitch(2) { _ =>
-                   Stream.fromEffect((latch.succeed(()) *> UIO.never).onInterrupt(substreamCancelled.set(true)))
-                 }
-                 .runDrain
-                 .either
-      cancelled <- substreamCancelled.get
-    } yield (cancelled must_=== true) and (result must beLeft("Ouch"))
-  }
-
-  private def flatMapParSwitchInnerDefectsInterruptAllFibers = unsafeRun {
-    val ex = new RuntimeException("Ouch")
-
-    for {
-      substreamCancelled <- Ref.make[Boolean](false)
-      latch              <- Promise.make[Nothing, Unit]
-      result <- Stream(
-                 Stream.fromEffect((latch.succeed(()) *> ZIO.never).onInterrupt(substreamCancelled.set(true))),
-                 Stream.fromEffect(latch.await *> ZIO.die(ex))
-               ).flatMapParSwitch(2)(identity)
-                 .run(Sink.drain)
-                 .run
-      cancelled <- substreamCancelled.get
-    } yield (cancelled must_=== true) and (result must_=== Exit.die(ex))
-  }
-
-  private def flatMapParSwitchOuterDefectsInterruptAllFibers = unsafeRun {
-    val ex = new RuntimeException()
-
-    for {
-      substreamCancelled <- Ref.make[Boolean](false)
-      latch              <- Promise.make[Nothing, Unit]
-      result <- (Stream(()) ++ Stream.fromEffect(latch.await *> ZIO.die(ex)))
-                 .flatMapParSwitch(2) { _ =>
-                   Stream.fromEffect((latch.s
```

Comment:
Btw not sure you had seen this? https://github.com/zio/zio/issues/1548
`MockClock` could be used to ensure deterministic timings.

Reply:
Yes I did, it's on my todo list 👍

Reply:
One thing to watch out for here is that the `MockEnvironment` provides the `MockClock` as its default `Clock` implementation. While the `MockClock` is great, it requires manually adjusting the time, so if you just drop in tests that previously relied on the live clock they are not likely to work. For tests that rely on the passage of time I would probably start by providing `Live[Clock]` to them to make sure that the test is working and has been migrated correctly and then look to move to the `MockClock`.

Reply:
oooh that's super useful, thanks! I'll do that and then we can look at `MockClock` in a follow-up. I tried using it when writing `zipWithLatest` but one fiber wasn't making progress, so I thought I was doing something wrong.